pkg/spec/storage.go

```diff
@@ -575,35 +579,52 @@ func ValidateVolumeCtrDir(ctrDir string) error {
 }
 
 // ValidateVolumeOpts validates a volume's options
-func ValidateVolumeOpts(options []string) error {
+func ValidateVolumeOpts(options []string) ([]string, error) {
 	var foundRootPropagation, foundRWRO, foundLabelChange, bindType int
+	finalOpts := make([]string, 0, len(options))
+	discardOpts := []string{"cached", "delegated"}
 	for _, opt := range options {
+		// The discarded ops are OS X specific volume options introduced
+		// in a recent Docker version.
+		// They have no meaning on Linux, so here we silently drop them.
+		// This matches Docker's behavior (the options are intended to
+		// be always safe to use, even not on OS X).
+		bad := false
```

Comment:
Nit: `s/bad/ignore/`

But somehow, I'd prefer to have that in the switch table below to let the compiler properly optimize.
```golang
case "cached", "delegated":
    // Comment ...
    continue
```

Reply:
I agree this seems like a simpler solution.

Reply:
I agree this seems like a simpler solution.

@QiWang19 @ashley-cui FYI

Reply:
Good catch. Pushed an updated version.